projects/dsp-ui/src/lib/viewer/operations/display-edit/display-edit.component.html

```diff
@@ -11,22 +11,23 @@ <h4 class="label">
-      <dsp-boolean-value class="parent-value-component" #displayVal *ngSwitchCase="constants.BooleanValue" [mode]="mode" [displayValue]=$any(displayValue)></dsp-boolean-value>
-      <dsp-uri-value class="parent-value-component" #displayVal *ngSwitchCase="constants.UriValue" [mode]="mode" [displayValue]=$any(displayValue)></dsp-uri-value>
-      <dsp-decimal-value class="parent-value-component" #displayVal *ngSwitchCase="constants.DecimalValue" [mode]="mode" [displayValue]=$any(displayValue)></dsp-decimal-value>
-      <dsp-color-value class="parent-value-component" #displayVal *ngSwitchCase="constants.ColorValue" [mode]="mode" [displayValue]=$any(displayValue)></dsp-color-value>
-      <dsp-interval-value class="parent-value-component" #displayVal *ngSwitchCase="constants.IntervalValue" [mode]="mode" [displayValue]=$any(displayValue)></dsp-interval-value>
-      <dsp-time-value class="parent-value-component" #displayVal *ngSwitchCase="constants.TimeValue" [mode]="mode" [displayValue]=$any(displayValue)></dsp-time-value>
-      <dsp-geoname-value class="parent-value-component" #displayVal *ngSwitchCase="constants.GeonameValue" [mode]="mode" [displayValue]=$any(displayValue)></dsp-geoname-value>
-      <dsp-link-value class="parent-value-component" #displayVal *ngSwitchCase="constants.LinkValue" [mode]="mode" [displayValue]=$any(displayValue)
+        <!-- display value is cast as 'any' because the compiler cannot infer the value type of the child component -->
+        <dsp-text-value-as-string class="parent-value-component" #displayVal *ngSwitchCase="'ReadTextValueAsString'" [mode]="mode" [displayValue]="$any(displayValue)"></dsp-text-value-as-string>
+        <dsp-text-value-as-html class="parent-value-component" #displayVal *ngSwitchCase="'ReadTextValueAsHtml'" [mode]="mode" [displayValue]="$any(displayValue)"></dsp-text-value-as-html>
+        <dsp-int-value class="parent-value-component" #displayVal *ngSwitchCase="constants.IntValue" [mode]="mode" [displayValue]="$any(displayValue)"></dsp-int-value>
+        <dsp-boolean-value class="parent-value-component" #displayVal *ngSwitchCase="constants.BooleanValue" [mode]="mode" [displayValue]="$any(displayValue)"></dsp-boolean-value>
+        <dsp-uri-value class="parent-value-component" #displayVal *ngSwitchCase="constants.UriValue" [mode]="mode" [displayValue]="$any(displayValue)"></dsp-uri-value>
+        <dsp-decimal-value class="parent-value-component" #displayVal *ngSwitchCase="constants.DecimalValue" [mode]="mode" [displayValue]="$any(displayValue)"></dsp-decimal-value>
+        <dsp-color-value class="parent-value-component" #displayVal *ngSwitchCase="constants.ColorValue" [mode]="mode" [displayValue]="$any(displayValue)"></dsp-color-value>
+        <dsp-interval-value class="parent-value-component" #displayVal *ngSwitchCase="constants.IntervalValue" [mode]="mode" [displayValue]="$any(displayVal)"></dsp-interval-value>
```

Comment:
$any(displayVal) ;-)

Reply:
Thanks! Fixed in 2f6f721

Reply:
`#displayVal` is actually a variable, this is why this compiled. But since it is wrapped with `$any()`, to compatibility check was performed.

https://angular.io/guide/template-syntax#template-reference-variables-var

Reply:
I wondered why the uni tests passed and checked the test in `display-edit.component.spec`:

```ts
it('should choose the apt component for an interval value in the template', () => {

      testHostComponent.assignValue('http://0.0.0.0:3333/ontology/0001/anything/v2#hasInterval');
      testHostFixture.detectChanges();

      expect(testHostComponent.displayEditValueComponent.displayValueComponent instanceof TestIntervalValueComponent).toBe(true);
      expect(testHostComponent.displayEditValueComponent.displayValueComponent.displayValue).not.toBeUndefined();
      expect(testHostComponent.displayEditValueComponent.displayValueComponent.mode).toEqual('read');
    });
```


`expect(testHostComponent.displayEditValueComponent.displayValueComponent.displayValue).not.toBeUndefined();` should actually check for an instance of a  `ReadIntervalValue`. That would make sure that the template correctly passed the `ReadIntervalValue` to the `ReadIntervalValueComponent`. The same applies to all other value types.

Shall we make a DSP task for this or do it in this PR?

Reply:
in case of `$any(displayVal)`  a reference to `TestIntervalValueComponent` was passed instead of the `ReadIn tervalValue`

Reply:
> I wondered why the unit tests passed and checked the test in `display-edit.component.spec`:
> 
> ```ts
> it('should choose the apt component for an interval value in the template', () => {
> 
>       testHostComponent.assignValue('http://0.0.0.0:3333/ontology/0001/anything/v2#hasInterval');
>       testHostFixture.detectChanges();
> 
>       expect(testHostComponent.displayEditValueComponent.displayValueComponent instanceof TestIntervalValueComponent).toBe(true);
>       expect(testHostComponent.displayEditValueComponent.displayValueComponent.displayValue).not.toBeUndefined();
>       expect(testHostComponent.displayEditValueComponent.displayValueComponent.mode).toEqual('read');
>     });
> ```
> 
> `expect(testHostComponent.displayEditValueComponent.displayValueComponent.displayValue).not.toBeUndefined();` should actually check for an instance of a `ReadIntervalValue`. That would make sure that the template correctly passed the `ReadIntervalValue` to the `ReadIntervalValueComponent`. The same applies to all other value types.
> 
> Shall we make a DSP task for this or do it in this PR?

Good find! Since not much has changed in this PR in the end, I'm okay making the changes in this PR

Reply:
<img width="1571" alt="Screen Shot 2020-06-12 at 09 25 19" src="https://user-images.githubusercontent.com/60604010/84476609-af069880-ac8e-11ea-8885-324306053956.png">

The test now correctly fails when using the wrong variable in the template